packages/providers/src/provider.ts

```diff
@@ -507,6 +509,40 @@ export default class Provider {
     }));
   }
 
+  /**
+   * Returns message for the given address
+   */
+  async getMessages(
+    /** The address to get message from */
+    address: AbstractAddress,
+    /** Pagination arguments */
+    paginationArgs?: CursorPaginationArgs
+  ): Promise<Message[]> {
+    const result = await this.operations.getMessages({
+      first: 10,
+      ...paginationArgs,
+      owner: address.toB256(),
+    });
+
+    const messages = result.messages.edges!.map((edge) => edge!.node!);
+
+    return messages.map((message) => {
+      const dataLength = message.data.length;
+      const [data] = new ArrayCoder(new NumberCoder('u8'), dataLength / WORD_SIZE).decode(
```

Comment:
Why do we need to divide data length by the `WORD_SIZE).decode(...`. Looking at the specs, it looks like data length

Reply:
Can we use the message decoder here?